tutor/utils.py

```diff
@@ -162,12 +163,29 @@ def docker(*command: str) -> int:
     return execute("docker", *command)
 
 
+@lru_cache
+def _which_docker_compose() -> str:
+    """
+    A helper function to determine which program to call when running docker compose
+    """
+    if os.environ.get("TUTOR_USE_COMPOSE_PLUGIN") is not None:
+        return "docker"
+    if shutil.which("docker-compose") is not None:
+        return "docker-compose"
+    if shutil.which("docker") is not None:
+        if subprocess.run(["docker", "compose"], capture_output=True).returncode == 0:
+            return "docker"
+    return "na"
+
+
 def docker_compose(*command: str) -> int:
-    if shutil.which("docker-compose") is None:
-        raise exceptions.TutorError(
-            "docker-compose is not installed. Please follow instructions from https://docs.docker.com/compose/install/"
-        )
-    return execute("docker-compose", *command)
+    if _which_docker_compose() == "docker":
+        return execute("docker", "compose", *command)
+    if _which_docker_compose() == "docker-compose":
```

Comment:
There are a few problems with your implementation:

1. Multiple calls are made to `_which_docker_compose`. Even though the results are lru-cached, it's inelegant.
2. The name of the `TUTOR_USE_COMPOSE_PLUGIN` environment variable is not really adequate. I understand that you refer to the Docker compose plugin, but people will mistake it with Tutor plugins (I did). I suggest instead ` TUTOR_USE_DOCKER_COMPOSE_V2`.
3. The string returned by `_which_docker_compose` is rather arbitrary and makes it more difficult to read the code.

I suggest that you move back the code of the `_which_docker_compose` back inline into the `docker_compose` function. You should only create a separate lru-cached function for fetching the `docker compose version`, which should return either a `str` or None.

Reply:
> I agree with you that the PLUGIN in the name seems confusing, but I really don't like to use V2 to refer to using docker compose as I've been using docker-compose v2 for a long time. So I propose TUTOR_USE_DC_AS_SUBCOMMAND.

"DC" is not easy to understand out of context. I counter-propose `TUTOR_USE_COMPOSE_SUBCOMMAND`.

> But the case is os.environ.get("TUTOR_USE_DC_AS_SUBCOMMAND") and shutil.which("docker-compose") gets computed over and over again. Wouldn't it be better to cache them?

Yes, you are right. Thus, I propose:

```
@lru_cache
def _docker_compose_command():
    if ...:
        return ("docker-compose",)
    if ...:
        return ("docker", "compose")
    raise ...

def docker_compose(*command):
    return execute(*_docker_compose_command(), *command)
```